lib/upgrade/migrator/migrate_test.go

```diff
@@ -57,3 +63,150 @@ var _ = Describe("UT for checking the version for migration.", func() {
 		Entry("Expect 1.2.3.4.5 to not migrate", "1.2.3.4.5", false, true),
 	)
 })
+
+type singleTypeClient struct {
+	kdd  bool
+	kvps []*model.KVPair
+}
+
+func (stc singleTypeClient) Apply(d *model.KVPair) (*model.KVPair, error) {
+	return nil, nil
+}
+
+func (stc singleTypeClient) Get(k model.Key) (*model.KVPair, error) {
+	ks := k.String()
+	for _, kvp := range stc.kvps {
+		if kvp.Key.String() == ks {
+			return kvp, nil
+		}
+	}
+	return nil, cerrors.ErrorResourceDoesNotExist{Identifier: k}
+}
+
+func (stc singleTypeClient) List(l model.ListInterface) ([]*model.KVPair, error) {
+	if _, ok := l.(model.ProfileListOptions); ok {
+		return stc.kvps, nil
+	}
+	return nil, nil
+}
+
+func (stc singleTypeClient) IsKDD() bool {
+	return stc.kdd
+}
+
+var _ = Describe("Test OpenStack migration filters", func() {
+
+	wk := model.WorkloadEndpointKey{
+		Hostname:       "ahost",
+		OrchestratorID: "orchestrator",
+		WorkloadID:     "wkid",
+		EndpointID:     "endID",
+	}
+	mac, _ := gnet.ParseMAC("ee:ee:ee:ee:ee:ee")
+	wv := model.WorkloadEndpoint{
+		State:            "Running",
+		Name:             "wepName",
+		ActiveInstanceID: "wepActInstID",
+		Mac:              &net.MAC{mac},
+		ProfileIDs:       []string{"wepProfIDs"},
+		IPv4Nets:         []net.IPNet{},
+		IPv6Nets:         []net.IPNet{},
+	}
+
+	It("should not filter WorkloadEndpoints without openstack as OrchestratorID", func() {
+		clientv1 := fakeClientV1{}
+
+		kvps := []*model.KVPair{
+			&model.KVPair{
+				Key:   wk,
+				Value: &wv,
+			},
+		}
+		clientv1.kvps = kvps
+
+		// Convert the data back to a set of resources.
+		mh := &migrationHelper{clientv1: clientv1}
+		data, err := mh.queryAndConvertResources()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(data.ConversionErrors).To(HaveLen(0))
+		By("Checking total conversion")
+		Expect(data.Resources).To(HaveLen(1))
```

Comment:
Perhaps commonize those lines into a utility routine whose only parameter is the expected number of converted resources?

Reply:
I don't understand the suggestion here, could you elaborate please?

Reply:
Bearing in mind that this set of lines is repeated 4 times, I just meant something like this:

```
func expectConvertedResources(client <whatever client's type is>, num int) {
    mh := &migrationHelper{clientv1: client}
    data, err := mh.queryAndConvertResources()
    Expect(err).NotTo(HaveOccurred())
    Expect(data.ConversionErrors).To(HaveLen(0))
    By("Checking total conversion")
    Expect(data.Resources).To(HaveLen(num))
}
```

Reply:
ah ok, I was thrown off by 'only parameter'.  Thanks